src/gregor/disaggregate.py

```diff
@@ -20,57 +23,162 @@ def disaggregate_polygon_to_raster(
+    debug: bool, default False
+        If True, print some info on what the process is doing to console.
 
     Returns
     -------
-    xr.Dataset
+    xr.DataArray
         Disaggregated raster data.
     """
-    _data = data.copy()
-    index_name = _data.index.name
-    if index_name is None:
-        index_name = "id"
-        _data.index.name = index_name
-
-    if not proxy.rio.crs == data.crs:
+    if isinstance(proxy, xr.DataArray):
+        proxy_da = proxy
+    else:  # proxy is a Dataset
+        if len(proxy.data_vars) == 1:
+            var_name = next(iter(proxy.data_vars))
+            proxy_da = proxy[var_name]
+        else:
+            raise ValueError(
+                f"Cannot compute multi-variable Dataset of length {len(proxy.data_vars)}. "
+                "Pass a DataArray instead."
+            )
+
+    gdf = data.copy()
+    index_name = gdf.index.name or "id"
+    gdf.index.name = index_name
+
+    if proxy.rio.crs != gdf.crs:
         print(
             f"CRS of `proxy` ({proxy.rio.crs}) does not match CRS of `data` ({data.crs}). Reprojecting CRS of `data` to `proxy`'s CRS."
         )
-        _data = _data.to_crs(proxy.rio.crs)
-
-    # Each raster point belongs to one spatial_unit
-    belongs_to = get_belongs_to_matrix(proxy, _data.geometry)
-    _data = _data[[column]].to_xarray()
-    normalization = proxy.groupby(belongs_to).sum().rename(group=index_name)
-
-    # # Remove regions that do not belong to any geometry
-    _data = _data.sel({index_name: normalization.coords[index_name]})
-
-    # Disaggregate data to raster using proxy
-    # raster_{x,y} = 1/normalization_{id} * _data_{id} * belongs_to_{id,x,y} * proxy_{x,y}
-    raster = xr.DataArray(data=0, dims=["y", "x"], coords={"y": proxy.y, "x": proxy.x})
-    for id in normalization.coords[index_name]:
-        raster_id = (
-            1
-            / normalization.sel({index_name: id})
-            * _data.sel({index_name: id})
-            * (belongs_to == id)
-            * proxy
-        )
-        raster = raster + raster_id
+        gdf = gdf.to_crs(proxy.rio.crs)
+
+    # always one DataArray in float32, chunked
+    proxy_da = proxy_da.astype("float32").chunk({"y": chunk_size, "x": chunk_size})
+
+    # raster of polygon IDs ─ always burn row numbers (0…n‑1)
+    if not np.issubdtype(gdf.index.dtype, np.integer):
+        print("Non-integer index detected, resetting to numeric for processing.")
+        geom_id_source = gdf.reset_index(drop=True).geometry  # integer IDs
+    else:
+        geom_id_source = gdf.geometry
+
+    belongs_to = get_belongs_to_matrix(proxy_da, geom_id_source)
+    if debug:
+        print("unique IDs in belongs_to:", np.unique(belongs_to.data)[:15])
+        print("min / max proxy value:", float(proxy_da.min()), float(proxy_da.max()))
+
+    belongs_to = belongs_to.where(~belongs_to.isnull(), other=-1)
+    belongs_to = belongs_to.astype("int32").chunk(proxy_da.chunks)
```

Comment:
Not necessary (except for the chunking), as this is done in get_belongs_to_matrix already.